kedro/framework/session/session.py

```diff
@@ -104,7 +105,9 @@ def __init__(  # noqa: PLR0913
         save_on_close: bool = False,
         conf_source: str | None = None,
     ):
-        self._project_path = Path(project_path or Path.cwd()).resolve()
+        self._project_path = Path(
+            project_path or _find_kedro_project(Path.cwd()).resolve() or Path.cwd()
```

Comment:
Previously we called `.resolve()` on all options `project_path` as well as `Path.cwd()` and now only on `_find_kedro_project(Path.cwd()).resolve()`, why did that change?

Reply:
Updated!